src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1395,7 +1400,7 @@ private bool resolveVirtualMethod(CORINFO_DEVIRTUALIZATION_INFO* info)
 
             if (info->pResolvedTokenVirtualMethod != null)
             {
-                methodWithTokenDecl = ComputeMethodWithToken(decl, ref *info->pResolvedTokenVirtualMethod, null, false);
+                methodWithTokenDecl = ComputeMethodWithToken(decl, ref *info->pResolvedTokenVirtualMethod, null, false, asyncVariant: false);
```

Comment:
I think the `asyncVariant: false` here are not quite correct

I would make this throw RequiresRuntimeJitException if `decl.IsAsync` or `impl.IsAsync`. for now.